sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -183,9 +194,17 @@ def from_blob_url(cls, blob_url, credential=None, snapshot=None, **kwargs):
 
         if not parsed_url.netloc:
             raise ValueError("Invalid URL: {}".format(blob_url))
-        account_url = parsed_url.netloc.rstrip('/') + "?" + parsed_url.query
-        path_blob = parsed_url.path.lstrip('/').partition('/')
-        container_name, blob_name = unquote(path_blob[0]), unquote(path_blob[2])
+
+        path_blob = parsed_url.path.lstrip('/').split('/')
+        account_path = ""
+        if len(path_blob) > 2:
+            account_path = "/" + "/".join(path_blob[:-2])
+        account_url = "{}://{}{}?{}".format(
+            parsed_url.scheme,
+            parsed_url.netloc.rstrip('/'),
+            account_path,
+            parsed_url.query)
+        container_name, blob_name = unquote(path_blob[-2]), unquote(path_blob[-1])
```

Comment:
is using the url from a portal a common way for user to paste a url? If that's the case we shouldn't do unquote here I guess?

Reply:
Interesting - this is not a change in behaviour - as we were always unquoting here...
Although I don't think it will do harm? Unless the blob name itself had a quote string in it, I don't think you can "double unquote"....

Reply:
Nevermind, I tried to copy the url from explorer, and the url is encoded. I think the way we got url from portal wasn't correct yesterday?

Reply:
Yeah, we cannot double unquote, just incase someone got a weird name for his file like %28.txt then after we unquote for them it is (

Reply:
Discussed offline - leaving as-is for now.